File: Software/workspace/CmdDiscusBot.demo/src/Robot.cpp

#include "Robot.h"
#include "CommandBase.h"
#include "Commands/Autonomous.h"


void Robot::RobotInit() {
	CommandBase::init();
	autonomousCommand = new Autonomous();
}

void Robot::AutonomousInit() {
	std::cout << "Starting Auto" << std::endl;
	autonomousCommand->Start();
}

void Robot::AutonomousPeriodic() {
	Scheduler::GetInstance()->Run();
}

void Robot::TeleopInit() {
	autonomousCommand->Cancel();
	std::cout << "Starting Teleop" << std::endl;
}

void Robot::TeleopPeriodic() {
	Scheduler::GetInstance()->Run();
}

void Robot::TestPeriodic() {
	lw->Run();
}

void Robot::DisabledInit(){
	std::cout << "DisabledInit" << std::endl;
	autonomousCommand->Cancel();
	CommandBase::drivetrain.get()->Reset();
	CommandBase::lifter.get()->Reset();

}

START_ROBOT_CLASS(Robot)

File: Software/workspace/JVisionThreadTestVSC/src/main/java/frc/robot/Robot.java
/*----------------------------------------------------------------------------*/
/* Copyright (c) 2017-2018 FIRST. All Rights Reserved.                        */
/* Open Source Software - may be modified and shared by FRC teams. The code   */
/* must be accompanied by the FIRST BSD license file in the root directory of */
/* the project.                                                               */
/*----------------------------------------------------------------------------*/
package frc.robot;
// note: need to set java.home in file->preferences->settings to avoid warnings
//       about unable to ressolve system classes (java.lang.object etc)

import edu.wpi.first.wpilibj.IterativeRobot;
import frc.robot.subsystems.VisionProcess;

/**
 * The VM is configured to automatically run this class, and to call the
 * functions corresponding to each mode, as described in the IterativeRobot
 * documentation. If you change the name of this class or the package after
 * creating this project, you must also update the build.gradle file in the
 * project.
 */
public class Robot extends IterativeRobot {
  /**
   * This function is run when the robot is first started up and should be used
   * for any initialization code.
   */
  VisionProcess vision;

  @Override
  public void robotInit() {
    vision = new VisionProcess();
    vision.init();
    vision.start();
  }

  @Override
  public void robotPeriodic() {
  }

  @Override
  public void autonomousInit() {

  }

  /**
   * This function is called periodically during autonomous.
   */
  @Override
  public void autonomousPeriodic() {

  }

  /**
   * This function is called periodically during operator control.
   */
  @Override
  public void teleopPeriodic() {
  }

  /**
   * This function is called periodically during test mode.
   */
  @Override
  public void testPeriodic() {
  }
}
